tests/posts_tests/test_posts.py

```diff
@@ -283,3 +285,142 @@ def test_several_pollfiles(self, persist_poll):
     def test_pollfile_invalid_args(self, poll, file, expected_error):
         with pytest.raises(expected_error):
             PollFile(poll=poll, file=file)
+
+
+# -------------------------------------------- Choice tests --------------------------------------------
+
+@pytest.mark.django_db()
+class TestChoice:
+
+    # Check that the Choice values are the same as the Choice inputs.
+    def test_new_choice_input_same_as_output(self, new_choice):
+        assert new_choice.poll.author.username == USERNAME
+        assert new_choice.poll.description == DESCRIPTION
+        assert new_choice.poll.get_amount_of_votes() == 0
+        assert new_choice.choice_text == CHOICETEXT1
+
+    # Check if the Choice is saved in the database and accessible via its Poll.
+    def test_persist_choice(self, persist_choice):
+        assert persist_choice in Choice.objects.all()
+        assert persist_choice in persist_choice.poll.choice_set.all()
+
+    # Check if Choice deletion delete only Choice from database.
+    def test_delete_choice(self, persist_choice):
+        persist_choice.delete()
+        assert persist_choice.poll in Poll.objects.all()
+        assert persist_choice.poll.author in User.objects.all()
+        assert persist_choice not in Choice.objects.all()
+
+    # Check if Choice's poll deletion delete both Choice and poll from database - but not the poll's author.
+    def test_delete_choice_poll(self, persist_choice):
+        persist_choice.poll.delete()
+        assert persist_choice.poll not in Poll.objects.all()
+        assert persist_choice.poll.author in User.objects.all()
+        assert persist_choice not in Choice.objects.all()
+
+    # Check if Choice's poll's author deletion delete also Choice and database.
+    def test_delete_choice_user(self, persist_choice):
+        persist_choice.poll.author.delete()
+        assert persist_choice not in Poll.objects.all()
+        assert persist_choice.poll.author not in User.objects.all()
+        assert persist_choice not in Choice.objects.all()
+
+    # Check if Poll can have several Choices simultaneously.
+    def test_several_choices(self, persist_poll):
+        choice1 = Choice(poll=persist_poll, choice_text=CHOICETEXT1)
+        assert len(Choice.objects.all()) == 0
+        choice1.save()
+        assert len(Choice.objects.all()) == 1
+        persist_poll.poll.choice_set.create(choice_text=CHOICETEXT2)
+        assert len(persist_poll.choice_set.all()) == 2
+
+    # verify that it is impossible to create a Choice with invalid params.
+    # (string as Poll, user as Poll, int as choice_text).
+    @pytest.mark.parametrize("poll, choice_text, expected_error", [(
+        USERNAME, CHOICETEXT1, ValueError),
+        (new_user, DESCRIPTION, ValueError),
+        (new_poll, 6, ValueError)])
+    def test_pollFile_invalid_args(self, poll, choice_text, expected_error):
+        with pytest.raises(expected_error):
+            PollFile(poll=poll, file=choice_text)
```

Comment:
Why do we try to create `PollFile`? \
Shouldn't it be `Choice(poll=poll, file=choice_text)`